v2/inc/c_logging/log_context.h

```diff
@@ -4,6 +4,241 @@
+            LOG_CONTEXT_PROPERTY_VALUE_PAIR* first_property_value_pair = destination_context.property_value_pairs_ptr; \
+            (void)property_value_pair; \
+            uint8_t* data_pos = destination_context.values_data; \
+            *data_pos = (uint8_t)(internal_log_context_get_property_value_pair_count_or_zero(parent_context) MU_IF(MU_COUNT_ARG(__VA_ARGS__), MU_FOR_EACH_1(COUNT_PROPERTY, __VA_ARGS__),)); \
+            /* Codes_SRS_LOG_CONTEXT_01_015: [ LOG_CONTEXT_LOCAL_DEFINE shall store one property/value pair that with a property type of struct with as many fields as the total number of properties passed to LOG_CONTEXT_LOCAL_DEFINE in the ... arguments. ]*/ \
+            first_property_value_pair->value = data_pos; \
+            first_property_value_pair->name = ""; \
+            first_property_value_pair->type = &struct_log_context_property_type; \
+            data_pos += 1 + internal_log_context_get_values_data_length_or_zero(parent_context); \
+            /* Codes_SRS_LOG_CONTEXT_01_016: [ LOG_CONTEXT_LOCAL_DEFINE shall store the property types and values specified by using LOG_CONTEXT_PROPERTY in the context. ]*/ \
+            MU_IF(MU_COUNT_ARG(__VA_ARGS__), MU_FOR_EACH_1(SETUP_PROPERTY_PAIR, __VA_ARGS__),) \
+        } \
+    } \
+
+LOG_CONTEXT_HANDLE log_context_create(LOG_CONTEXT_HANDLE parent_context, uint32_t properties_count, uint32_t data_size);
+void log_context_destroy(LOG_CONTEXT_HANDLE log_context);
+
+// macro that can be used to create a dynamically allocated context
+#define LOG_CONTEXT_CREATE(destination_context, parent_context, ...) \
+    { \
+        destination_context = log_context_create(parent_context, internal_log_context_get_property_value_pair_count_or_zero(parent_context) + 1 MU_IF(MU_COUNT_ARG(__VA_ARGS__), MU_FOR_EACH_1(COUNT_PROPERTY, __VA_ARGS__),), internal_log_context_get_values_data_length_or_zero(parent_context) + 1 MU_IF(MU_COUNT_ARG(__VA_ARGS__), MU_FOR_EACH_1(COUNT_DATA_BYTES, __VA_ARGS__),)); \
+        MU_IF(MU_COUNT_ARG(__VA_ARGS__), LOG_CONTEXT_CHECK_VARIABLE_ARGS(__VA_ARGS__),) \
+        if (destination_context != NULL) \
+        { \
+            /* Codes_SRS_LOG_CONTEXT_01_003: [ LOG_CONTEXT_CREATE shall store the property types and values specified by using LOG_CONTEXT_PROPERTY in the context. ]*/ \
+            LOG_CONTEXT_PROPERTY_VALUE_PAIR* property_value_pair = destination_context->property_value_pairs_ptr + internal_log_context_get_property_value_pair_count_or_zero(parent_context) + 1; \
+            LOG_CONTEXT_PROPERTY_VALUE_PAIR* first_property_value_pair = destination_context->property_value_pairs_ptr; \
+            (void)property_value_pair; \
+            uint8_t* data_pos = destination_context->values_data; \
+            *data_pos = (uint8_t)(internal_log_context_get_property_value_pair_count_or_zero(parent_context) MU_IF(MU_COUNT_ARG(__VA_ARGS__), MU_FOR_EACH_1(COUNT_PROPERTY, __VA_ARGS__),)); \
```

Comment:
I'm a little confused by this line here.  It sets data_pos to values_data, then immediately set it again.  Why set it to values_data in the first place if it just gets overwritten.  Am I missing something?

Reply:
In ln 213 the pointer value is computed. In ln 214 we actually write the data into the address pointed to by data_pos